tests/neuron/test_rotary_embedding.py

```diff
@@ -0,0 +1,55 @@
+# SPDX-License-Identifier: Apache-2.0
+"""
+Tests for miscellaneous utilities
+"""
+
+import pytest
+import torch
+
+from vllm.model_executor.layers.rotary_embedding import RotaryEmbedding
+from vllm.platforms import current_platform
+
+
+@pytest.mark.parametrize(
+    "max_position,is_neox_style,rotary_dim,head_size,seq_len", [
```

Comment:
We should add a test case where `rotary_dim != head_size`

Reply:
added related test case, along with function implementation.